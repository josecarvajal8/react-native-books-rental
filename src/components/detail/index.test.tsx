import React from 'react';
import renderer from 'react-test-renderer';
import Detail from '.';

test('Render detail component correctly', ()=> {
    const detailCmp = renderer.create(<Detail navigation={null}/>).toJSON();
    expect(detailCmp).toMatchSnapshot();
})