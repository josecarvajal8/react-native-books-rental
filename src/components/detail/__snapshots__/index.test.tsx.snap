// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render detail component correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "height": 100,
        "width": "100%",
        "zIndex": 100,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/img/bc_nav_bar_3x.png",
        }
      }
      style={
        Object {
          "height": 200,
          "position": "absolute",
          "top": -45,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 5,
          "paddingHorizontal": 5,
          "zIndex": 200,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
      >
        <View
          accessible={true}
          style={
            Object {
              "height": 20,
              "opacity": 1,
              "width": 20,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ÔÅì
          </Text>
        </View>
      </RNGestureHandlerButton>
      <Text
        forwardedRef={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 25,
          }
        }
      >
        [missing "en.titles.detail" translation]
      </Text>
      <View
        style={
          Object {
            "width": 20,
          }
        }
      />
    </View>
  </RCTSafeAreaView>,
  <RCTScrollView
    bounces={false}
    style={
      Object {
        "backgroundColor": "#dee7ea",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 8,
              "elevation": 10,
              "height": 300,
              "marginTop": 20,
              "shadowColor": "rgba(46, 48, 48, 1)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 2,
              "width": "90%",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "",
                }
              }
              style={
                Object {
                  "height": 130,
                  "marginLeft": 30,
                  "marginTop": 20,
                  "width": 100,
                }
              }
            />
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginTop": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 20,
                  }
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#626565",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 15,
                    "marginTop": 2,
                  }
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#626565",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 15,
                    "marginTop": 2,
                  }
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#626565",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 15,
                    "marginTop": 2,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "marginLeft": 30,
              }
            }
          >
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
            >
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ffffff",
                    "borderColor": "#0baee6",
                    "borderRadius": 30,
                    "borderWidth": 1,
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 30,
                    "opacity": 1,
                    "padding": 10,
                    "width": 300,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#0baee6",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 18,
                    }
                  }
                >
                  [MISSING "EN.DETAIL.ADDTOWISHLIST" TRANSLATION]
                </Text>
              </View>
            </RNGestureHandlerButton>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
            >
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#0baee6",
                    "borderRadius": 30,
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 15,
                    "opacity": 1,
                    "padding": 10,
                    "width": 300,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 18,
                    }
                  }
                >
                  [MISSING "EN.DETAIL.RENT" TRANSLATION]
                </Text>
              </View>
            </RNGestureHandlerButton>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
